components/core/src/clp_s/log_converter/LogSerializer.hpp

```diff
@@ -0,0 +1,109 @@
+#ifndef CLP_S_LOG_CONVERTER_LOGSERIALIZER_HPP
+#define CLP_S_LOG_CONVERTER_LOGSERIALIZER_HPP
+
+#include <cstddef>
+#include <optional>
+#include <string_view>
+#include <utility>
+
+#include <ystdlib/error_handling/Result.hpp>
+
+#include "../../clp/ffi/ir_stream/protocol_constants.hpp"
+#include "../../clp/ffi/ir_stream/Serializer.hpp"
+#include "../../clp/ir/types.hpp"
+#include "../../clp/type_utils.hpp"
+#include "../FileWriter.hpp"
+
+namespace clp_s::log_converter {
+/**
+ * Utility class that generates KV-IR corresponding to a converted input file.
+ */
+class LogSerializer {
+public:
+    // Constructors
+    // Disable copy constructor/assignment operator
```

Comment:
```suggestion
    // Delete copy constructor and assignment operator
```
According to the internal guideline: https://www.notion.so/yscope/WIP-Coding-Guidelines-9a308b847a5343958ba3cb97a850be66?source=copy_link#e7dbc12f0f4f470fa31d63043ee4c0c9